multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/helpers/v2/ConfigurationReferencesResolver.java

```diff
@@ -121,7 +122,12 @@ protected List<RequiredDependency> expandRequiredDependencyIfNecessary(Propertie
         }
 
         if (!permitsMultipleResources(dependency)) {
-            makeSureIsResolvedToSingleResource(dependency.getName(), resolvedReference.getResolvedResources());
+            if (dependencyOwner instanceof Resource) {
+                Resource resource = MiscUtil.cast(dependencyOwner);
+                makeSureIsResolvedToSingleResource(dependency.getName(), resolvedReference.getResolvedResources(), resource.isOptional());
+            } else {
+                makeSureIsResolvedToSingleResource(dependency.getName(), resolvedReference.getResolvedResources(), false);
+            }
```

Comment:
This check is not required here as optional resources are part of v3 schema, this code will never be executed